parsers/CO.py

```diff
@@ -74,9 +74,15 @@ def fetch_live_consumption(
     demand_list = TotalConsumptionList(logger)
 
     for datapoint in demand_data:
+        # Ensure consistent millisecond precision for ISO compliance:
+        # Checks if the 'Fecha' field contains a decimal point (indicating milliseconds)
+        # and if the milliseconds portion is less than three digits. If so, it pads the 'Fecha'
+        # value with zeros to achieve the required ISO standard precision.
+        if ("." in datapoint["Fecha"]) and (len(datapoint["Fecha"].split(".")[1]) != 3):
+            datapoint["Fecha"] = datapoint["Fecha"].ljust(23, "0")
         demand_list.append(
             zoneKey=zone_key,
-            datetime=arrow.get(datapoint["Fecha"], tzinfo=TZ).datetime,
+            datetime=datetime.fromisoformat(datapoint["Fecha"]).astimezone(tz=TZ),
```

Comment:
I would probably use replace here but I think both should do the same.

As long as the output is what you expect this is fine.